### DISCUSSION TITLE
Creating a single executable

### DISCUSSION THREAD

**Discussion by adammetcalf** [OpenDDS/OpenDDS#4726] [Q&A] [answered]
Hello,

This is on windows (x64 target).

I have worked through the 'getting started' guide and have compiled everything needed to run a publisher (as an executable) and a subscriber (as an executable). Both use the same rtps.ini file for discovery.

However, to run each executable I must first launch an x64 native vs19 command prompt, cd to my opendds directory and run setenv.bat to set the environment variables. I can then call the exe from the same window. What would you recommend I do so that I can simply run each executable from a fresh prompt without the requirement to set the environment variables every time?

**Comment by mitza-oci** [comment, marked as answer]
The full development environment is not needed when running applications.  Just make sure that the PATH has all the `*.dll` files that the process will need to load.  OpenDDS also supports a static library build where all ACE/TAO/OpenDDS code is built in to the executable file.

**Reply by adammetcalf** [reply]
Thanks very much! I appreciate the super quick response.

Just to clarify, is this something that should be declared in the mpc file?
